lxd/patches.go

```diff
@@ -245,58 +247,150 @@ func patchStorageApi(name string, d *Daemon) error {
 	// appropriate device including a pool is added to the default profile
 	// or the user explicitly passes the pool the container's storage volume
 	// is supposed to be created on.
-	defaultID, defaultProfile, err := dbProfileGet(d.db, "default")
+	profiles, err := dbProfiles(d.db)
 	if err == nil {
-		foundRoot := false
-		for k, v := range defaultProfile.Devices {
-			if v["type"] == "disk" && v["path"] == "/" && v["source"] == "" {
-				// Add the default storage pool.
-				defaultProfile.Devices[k]["pool"] = poolName
-				foundRoot = true
+		for _, pName := range profiles {
+			pID, p, err := dbProfileGet(d.db, pName)
+			if err != nil {
+				// Do not fail.
+				continue
```

Comment:
Actually, there's no reason why dbProfileGet would fail, so if that fails, it means the DB is completely broken and we should fail.